src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,231 @@
+    InterruptedException {
+
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+    result.iterateAll().forEach(value -> bigQueryRows.add(value.get(0).getValue()));
+  }
+
+  public static TableResult getTableNamesFromDataSet() throws IOException, InterruptedException {
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT table_name FROM `" + projectId + "." + dataset + "`.INFORMATION_SCHEMA.TABLES ";
+
+    return BigQueryClient.getQueryResult(selectQuery);
+  }
+
+  public static String getTableByName() throws IOException, InterruptedException {
+    String tableName = PluginPropertyUtils.pluginProp("bqTargetTable");
+    TableResult tableResult = getTableNamesFromDataSet();
+    Iterable<FieldValueList> rows = tableResult.iterateAll();
+
+    for (FieldValueList row : rows) {
+      FieldValue fieldValue = row.get(0);
+      String currentTableName = fieldValue.getStringValue();
+
+      if (currentTableName.equals(tableName)) {
+        return currentTableName;
+      }
+    }
+
+    return null; // Table not found
+  }
+
+  public static boolean compareSalesforceAndJsonData(List<JsonObject> salesforceData, List<JsonObject> bigQueryData,
+                                                     String tableName) {
+    boolean result = false;
+    if (bigQueryData == null) {
+      Assert.fail("bigQueryData is null");
+      return result;
+    }
+
+    int columnCountTarget = 0;
+    if (salesforceData.size() > 0) {
+      columnCountTarget = salesforceData.get(0).entrySet().size();
+    }
+    // Get the column count of the first JsonObject in bigQueryData
+    int jsonObjectIdx = 0;
+    int columnCountSource = 0;
+    if (bigQueryData.size() > 0) {
+      columnCountSource = bigQueryData.get(jsonObjectIdx).entrySet().size();
+    }
+
+    BigQuery bigQuery = BigQueryOptions.getDefaultInstance().getService();
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+
+    // Build the table reference
+    TableId tableRef = TableId.of(projectId, dataset, tableName);
+
+    // Get the table schema
+    Schema schema = bigQuery.getTable(tableRef).getDefinition().getSchema();
+
+    // Iterate over the fields
+    int currentColumnCount = 1;
+    while (currentColumnCount <= columnCountSource) {
+      for (Field field : schema.getFields()) {
+        String columnName = field.getName();
+        String columnType = field.getType().toString();
+        String columnTypeName = field.getType().getStandardType().name();
```

Comment:
Its unused I believe, remove it if its not used.

Reply:
done